pinot-integration-tests/src/test/java/org/apache/pinot/compat/tests/StreamOp.java

```diff
@@ -71,18 +125,170 @@ public void setInputDataFileName(String inputDataFileName) {
     _inputDataFileName = inputDataFileName;
   }
 
+  public String getTableConfigFileName() {
+    return _tableConfigFileName;
+  }
+
+  public void setTableConfigFileName(String tableConfigFileName) {
+    _tableConfigFileName = tableConfigFileName;
+  }
+
+  public String getRecordReaderConfigFileName() {
+    return _recordReaderConfigFileName;
+  }
+
+  public void setRecordReaderConfigFileName(String recordReaderConfigFileName) {
+    _recordReaderConfigFileName = recordReaderConfigFileName;
+  }
+
   @Override
   boolean runOp() {
-    System.out.println("Produce rows into stream " + _streamConfigFileName + " and verify rows in tables "
-        + _tableConfigFileNames);
+    switch(_op) {
+      case CREATE:
+        return createKafkaTopic();
+      case PRODUCE:
+        return produceData();
+    }
     return true;
   }
 
-  public List<String> getTableConfigFileNames() {
-    return _tableConfigFileNames;
+  private boolean createKafkaTopic() {
+    try {
+      Properties streamConfigMap = JsonUtils.fileToObject(new File(_streamConfigFileName), Properties.class);
+      String topicName = streamConfigMap.getProperty(TOPIC_NAME);
+      int partitions = Integer.parseInt(streamConfigMap.getProperty(NUM_PARTITIONS));
+
+      final Map<String, Object> config = new HashMap<>();
+      config.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, ClusterDescriptor.DEFAULT_HOST + ":" + ClusterDescriptor.KAFKA_PORT);
+      config.put(AdminClientConfig.CLIENT_ID_CONFIG, "Kafka2AdminClient-" + UUID.randomUUID().toString());
+      config.put(AdminClientConfig.REQUEST_TIMEOUT_MS_CONFIG, 15000);
+      AdminClient adminClient = KafkaAdminClient.create(config);
+      NewTopic topic = new NewTopic(topicName, partitions, (short) 1);
```

Comment:
Can you define this 1 as `private static final short KAFKA_REPLICATION_FACTOR` ?